src/main/java/redis/clients/jedis/util/RedisInputStream.java

```diff
@@ -43,9 +43,9 @@ public RedisInputStream(InputStream in) {
     this(in, INPUT_BUFFER_SIZE);
   }
 
-  public Byte peekByte() {
-    ensureFillSafe();
-    return buf[count];
+  public boolean peek(byte b) throws JedisConnectionException {
+    ensureFill(); // in current design, at least one reply is expected. so ensureFillSafe() is not necessary.
```

Comment:
But, what happens if there is no reply? It shouldn't occur, but maybe there's something we can do for resiliency here. At the very least raise a data specific exception so that we can know this happened?

Reply:
@chayim A JedisConnectionException wrapping a ReadTimeoutException will be thrown.